libs/article-parser/src/lib/correlation-score/correlation-score.ts

```diff
@@ -77,16 +111,84 @@ function getParagraphCorrelationScore(
   };
 }
 
+/**
+ * Get the correlation score for the segment of a paragraph with the most matches
+ * Will shorten the paragraph as much as possible while trying to mantain the same correlation score
+ * + or - {@code maintainWithinPercent}
+ */
+function getShortestParagraphCorrelationScore(
+  paragraph: string,
+  impacted: string,
+  recommendation: string,
+  impactedSynonyms: string[],
+  recommendationSynonyms: string[],
+  maintainWithinPercent = cutOffs.maintainScoreWithinPercent
+): ParsedArticleParagraph {
+  function calculatePercentageDifference(x: number, y: number): number {
+    return Math.abs((x - y) / x) * 100;
+  }
+  const sentences = paragraph.split('.');
+  // remove the last element if it is empty
+  if (!sentences[sentences.length - 1]) {
+    sentences.pop();
+  }
+  const initScore = getWholeParagraphCorrelationScore(
+    paragraph,
+    impacted,
+    recommendation,
+    impactedSynonyms,
+    recommendationSynonyms
+  ).correlationScore;
+  let currentScore = initScore;
+  let leftInd = 0;
+  let rightIndNonInclusive = sentences.length;
+  while (
+    calculatePercentageDifference(initScore, currentScore) <=
+      maintainWithinPercent / 2 &&
+    leftInd != rightIndNonInclusive
+  ) {
+    currentScore = getWholeParagraphCorrelationScore(
+      sentences.slice(leftInd, rightIndNonInclusive).join('.'),
+      impacted,
+      recommendation,
+      impactedSynonyms,
+      recommendationSynonyms
+    ).correlationScore;
+    leftInd++;
+  }
+  while (
+    calculatePercentageDifference(initScore, currentScore) <=
+      maintainWithinPercent &&
+    leftInd != rightIndNonInclusive
+  ) {
+    currentScore = getWholeParagraphCorrelationScore(
```

Comment:
wont' this break out of the while loop right *after* the current score is just out of range of maintainWithinPercent, so our currentScore that we select will be technically out of expected range?

Reply:
Oh yeah, so it would have to be rightIndNonInclusive + 1

Reply:
Iff rightIndNonInclusive != sentences.length